src/q2g-con-hypercube-qvx/TableFunc.cs

```diff
@@ -84,64 +83,56 @@ public TableHelper GetTableInfosFromApp(string tableName, ScriptCode script, Con
+                if (app == null)
+                    throw new Exception("No App session.");
 
-                if (qlikApp == null)
-                    return null;
+                var tableObject = app.GetObjectAsync(script.ObjectId).Result;
+                if (tableObject == null)
+                    throw new Exception("No Table object found.");
+                logger.Debug($"TableObject objectId: {script.ObjectId} - objectType {tableObject.qGenericType}");
 
-                var masterObject = qlikApp.FirstSession.CurrentApp.GetMasterObjectAsync(script.ObjectId).Result;
-                if (masterObject != null)
-                {
-                    logger.Debug($"find master object: {script.ObjectId}");
-                    var genericObject = qlikApp.FirstSession.CurrentApp.CreateGenericSessionObjectAsync(masterObject.Properties).Result;
-                    pager = genericObject.GetAllHyperCubePagers().FirstOrDefault() ?? null;
-                    var tableLayout = genericObject.GetLayout().As<TableLayout>();
-                    var hyperCube = tableLayout.HyperCube;
-                    columnOrder = hyperCube.ColumnOrder;
-                    size = hyperCube.Size;
-                    fields.AddRange(GetHyperCubeFields(hyperCube.DimensionInfo, hyperCube.MeasureInfo, script));
-                }
-                var table = qlikApp.FirstSession.CurrentApp.GetObjectAsync<Table>(script.ObjectId).Result;
-                if (table != null)
-                {
-                    logger.Debug($"table object: {script.ObjectId}");
-                    pager = table.HyperCubePager;
-                    columnOrder = table.ColumnOrder;
-                    size = table.Size;
-                    fields.AddRange(GetHyperCubeFields(table.DimensionInfo, table.MeasureInfo, script));
-                }
+                dynamic hyperCubeLayout = tableObject.GetLayoutAsync<JObject>().Result;
+                HyperCube hyperCube = hyperCubeLayout.qHyperCube.ToObject<HyperCube>();
+                columnOrder = hyperCube.qColumnOrder;
+                size = hyperCube.qSize;
+                fields.AddRange(GetHyperCubeFields(hyperCube.qDimensionInfo, hyperCube.qMeasureInfo, script));
 
                 var preview = new PreviewResponse()
                 {
-                     MaxCount = 15,
+                    MaxCount = 15,
                 };
 
                 if (script != null)
                 {
-                    var initalPage = new NxPage { Top = 0, Left = 0, Width = size.cx, Height = preview.MaxCount };
-                    var allPages = new List<IEnumerable<NxDataPage>>();
-                    allPages.Add(pager.GetData(new List<NxPage>() { initalPage }));
+                    var initalPage = new NxPage { qTop = 0, qLeft = 0, qWidth = size.qcx, qHeight = preview.MaxCount };
+                    var pages = tableObject.GetHyperCubeDataAsync("/qHyperCubeDef", new List<NxPage>() { initalPage }).Result;

```

Comment:
@mossbeard 
I don't understand that fetch. You don't have to fetch an InitilaPage and than the pages, you can just fetch alls pages, 
you just have to calculate how many pages you have to fetch, to divide the data.